themes/default/content/what-is/what-is-google-cloud-secret-manager.md

```diff
@@ -0,0 +1,113 @@
+
+You will then need to install the [gcloud CLI](https://cloud.google.com/cli). Once you have installed the gcloud CLI, run the `gcloud auth login` command to [authenticate to your Google Cloud account](https://cloud.google.com/sdk/gcloud/reference/auth/login).
+
+```bash
+# Example gcloud auth login interaction and output
+$ gcloud auth login
+Go to the following link in your browser:
+
+    https://accounts.google.com/o/oauth2/auth?...
+
+Enter authorization code: 4/0A....
+
+You are now logged in as [your@email.com].
+Your current project is [your-project].  You can change this setting by running:
+  $ gcloud config set project PROJECT_ID
+```
+
+### Create a secret via the CLI
+
+You can create a secret by running the `gcloud secrets create` command:
+
+```bash
+$ gcloud secrets create my-secret
+
+Created secret [my-secret].
+```
+
+Now add your secret data to the secret as shown below:
+
+```bash
+$ echo -n "my-secret-data" | gcloud secrets versions add my-secret --data-file=-
+
+Created version [1] of the secret [my-secret].
+```
+
+Verify the secret was created with the `gcloud secrets list` command.
+
+```bash
+$ gcloud secrets list
+
+NAME         CREATED              REPLICATION_POLICY  LOCATIONS
+api-key      2023-10-09T09:04:30  automatic           -
+my-secret    2023-11-30T14:05:58  automatic           -
+test-secret  2023-11-10T09:40:29  automatic           -
+```
+
+{{< notes type="info" >}}
+
+Many infrastructure as code platforms, including Pulumi, have support for creating secrets. You can learn more about how to create and manage secrets in Pulumi by taking a look at [Pulumi Secrets documentation](/docs/concepts/secrets/).
+
+{{< /notes >}}
+
+## Accessing Google Cloud Secret Manager secrets
+
+Now that you have created a Google Cloud Secret Manager secret, you can access the value via the gcloud CLI using the `gcloud secrets versions access` command.
+
+```bash
+$ gcloud secrets versions access latest --secret="my-secret"
+
+my-secret-data
+```
+
+## Challenges and considerations
+
+Google Cloud Secret Manager, while a powerful tool for managing secrets and cryptographic keys, does come with its own set of challenges, considerations, and limitations. Some of the key aspects to be aware of include:
+
+- **Compatibility and integration**: Google Cloud Secret Manager is optimized for workloads in Google Cloud. For organizations operating in a multi-cloud or hybrid cloud environment, integrating Secret Manager with other cloud platforms might not be straightforward. In such cases, using a third-party tool like [Pulumi ESC](/docs/pulumi-cloud/esc/) might be more effective. For example, Pulumi ESC provides a unified secret management interface across different cloud environments, simplifying the management of secrets in a multi-cloud setup.
```

Comment:
just want to call out this is the only place you drop neutrality on just expalining Secret Manager and pitch ESC as opposed to pitching it at the end

Reply:
This was added to align with your feedback about the goal being to funnel users into meatier Pulumi topics.

Reply:
Remove ESC reference/link